be/src/vec/olap/block_reader.cpp

```diff
@@ -84,7 +84,8 @@ void BlockReader::_init_agg_state() {
 
     auto& tablet_schema = tablet()->tablet_schema();
     for (auto idx : _agg_columns_idx) {
-        FieldAggregationMethod agg_method = tablet_schema.column(idx).aggregation();
+        FieldAggregationMethod agg_method =
+                tablet_schema.column(_real_columns_idx[idx]).aggregation();
```

Comment:
_real_columns_idx[idx] 
can be replaced by 
read_params.origin_return_columns->at(_return_columns_loc[idx])

Reply:
Okay,I modified this.

Reply:
`read_params` has been gone after init().

I think it is better to named `_real_agg_columns_idx`  or `_origin_agg_columns_idx`